### PR TITLE
first attempt to convert async-waitany msg version to s4u (doesn't work yet)

### PR DESCRIPTION
**Pull request by Takishipp**

### CODE REVIEW

File: examples/s4u/async-waitany/s4u_async-waitany.cpp [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    snprintf(taskname,79, "Task_%d", i);
+
+    comm = mbox->put_async((void*)taskname, 42.0);
+    comms.push_back(comm);
+    XBT_INFO("Send to receiver-%ld %s comm_size %f", i % receivers_count, taskname, task_comm_size / coef);
+  }
+
+  /* Here we are waiting for the completion of all communications */
+  while (!comms.empty()) {
+    comms.pop_back();
```
Comment (mquinson) [inline examples/s4u/async-waitany/s4u_async-waitany.cpp]: You are missing a call to comm->wait() here

File: examples/s4u/async-waitany/s4u_async-waitany.cpp [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    comm = mbox->get_async(&received);
+    comms.push_back(comm);
+  }
+
+  /* Here we are waiting for the receiving of all communications */
+  while (!comms.empty()) {
+    // returns the rank of the comm that just ended. Remove it.
+    comm=comms.back();
+    comms.pop_back();
+    // simgrid::s4u::this_actor::execute(comm);
```
Comment (mquinson) [inline examples/s4u/async-waitany/s4u_async-waitany.cpp]: Remove this comment and add a comm->wait() instead

File: examples/s4u/async-waitany/s4u_async-waitany.cpp [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+}
+
+int main(int argc, char *argv[])
+{
+  simgrid::s4u::Engine *e = new simgrid::s4u::Engine(&argc,argv);  
+
+  e->loadDeployment(argv[2]); 
+ 
+  e->registerFunction<sender>("sender");   
+  // e->registerFunction<receiver>("receiver");
```
Comment (mquinson) [inline examples/s4u/async-waitany/s4u_async-waitany.cpp]: Why did you comment this?

File: examples/s4u/async-waitany/s4u_async-waitany.tesh [hunk 4]
```diff
@@ -0,0 +1,34 @@
+#! ./tesh
+
+p Testing the MSG_comm_waitany function
+
+! output sort 19
+$ $SG_TEST_EXENV ${bindir:=.}/async-waitany ${srcdir:=.}/small_platform.xml ${srcdir:=.}/../msg/async-waitany/async-waitany_d.xml "--log=root.fmt:[%10.6r]%e(%i:%P@%h)%e%m%n"
```
Comment (mquinson) [inline examples/s4u/async-waitany/s4u_async-waitany.tesh]: Is this test working as expected?

**Comment by mquinson** [comment]
Also, you should fix the errors reported by codacy. And of course, your code is supposed to at least compile before you commit it. Having broken tests is not good, but committing something that does not compile is not the way to go.

Cheers, Mt.

File: examples/s4u/async-waitany/s4u_async-waitany.cpp (hunk 1 above)
Comment (Takishipp) [reply examples/s4u/async-waitany/s4u_async-waitany.cpp]: I think that we don't have to put wait over here.

**Comment by mquinson** [comment]
Please, please, please, DO NOT merge more than one example per pull request. It only complicates thing.

**Comment by mquinson** [comment]
Thanks for this. Next time, please make sure that it passes all the tests !

**Pull request merged**